### PR TITLE
Add toString() for better resource debugging

### PR DESCRIPTION
**Pull request by iloveeclipse**
See https://github.com/eclipse-platform/eclipse.platform/issues/470

### CODE REVIEW

**Comment by fedejeanne** [comment]
Could the calls to methods like `exists()` or `isSet(int)` become a bottleneck or are they mere bitwise comparisons?

**Comment by iloveeclipse** [comment]
It's toString(), which bottleneck you mean? It's only used while debugging code.

**Comment by fedejeanne** [comment]
I know, debugging is also a use-case for us developers, that's why I ask. Additionally, one can never completely rule out the possibility that production code is using `toString()` too (even if it's a bad idea). 

Let me rephrase :-)
I've noticed that `toString()` now call methods as opposed to just reading from fields. I trust they will not (significantly) degrade performance, even when debugging? I can live with a slight performance decrease when debugging if it helps to debug.

**Comment by iloveeclipse** [comment]
No, they don't degrade performance in a way anyone would notice that.

**Comment by iloveeclipse** [comment]
Test failis unrelated.
